flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -411,6 +411,23 @@ class TaskManager(
             log.debug(s"Cannot find task to fail for execution ${executionID})")
           }
 
+        // stops a task
+        case StopTask(executionID) =>
+          val task = runningTasks.get(executionID)
+          if (task != null) {
+            try {
+              task.stopExecution()
+              sender ! new TaskOperationResult(executionID, true)
+            } catch {
+              case t: Throwable =>
+                        sender ! new TaskOperationResult(executionID, false, t.getMessage())
```

Comment:
Hmmm... `task.stopExecution()` has the call to `userfunction.cancel()` nested inside. If there is an exception there it can hit us here... So catching `Throwable` an send back the result using `StringUtils.stringifyException` might just be fine.